repos/exporter/lib/atcoCheckerHandler.ts

```diff
@@ -0,0 +1,286 @@
+            const idsOfServicesRequiringAttention: string[] = await getIdsOfServicesRequiringAttentionForReturns(
+                pointToPointTicket,
+            );
+
+            await saveIdsOfServicesRequiringAttentionInTheDb(productId, idsOfServicesRequiringAttention);
+        }
+    }
+};
+
+/**
+ * For single tickets; gets a list of ids for the serices that require
+ * attention because of a change on the stops relating to the service.
+ *
+ * @param {pointToPointTicket} the matching json for single ticket.
+ *
+ * @returns {Promise<string[]>} a string array containing the ids of the
+ * services that require attention.
+ */
+const getIdsOfServicesRequiringAttentionForSingles = async (
+    pointToPointTicket: WithIds<SingleTicket>,
+): Promise<string[]> => {
+    // all stops from the matching JSON
+    let atcoCodesOfKnownStops: string[] = [];
+
+    pointToPointTicket.fareZones.forEach((fareZone) => {
+        fareZone.stops.forEach((stop) => {
+            atcoCodesOfKnownStops.push(stop.atcoCode);
+        });
+    });
+
+    if (pointToPointTicket.unassignedStops.singleUnassignedStops) {
+        pointToPointTicket.unassignedStops.singleUnassignedStops.forEach((stop) => {
+            atcoCodesOfKnownStops.push(stop.atcoCode);
+        });
+    }
+
+    // by this point, atcoCodesOfKnownStops is populated.
+
+    const { journeyDirection, lineId, nocCode } = pointToPointTicket;
+
+    const directionsAndStops = await getDirectionAndStopsByLineIdAndNoc(lineId, nocCode);
+
+    const theStopsMatchingTheJourneyDirectionOnOurTicket = directionsAndStops.filter(
+        (directionsAndStopsItem) => directionsAndStopsItem.direction === journeyDirection,
+    );
+
+    // split by service id and build a hashmap (dictionary) where the key is
+    // the service id and the value is the array of stops
+    const serviceAndStopsFromFeed = buildAHashmapSplitOnServiceId(theStopsMatchingTheJourneyDirectionOnOurTicket);
+
+    // sort the stops and remove duplicates for each service
+    const serviceAndSortedUniqueStopsFromFeed = sortAndRemoveDuplicateStops(serviceAndStopsFromFeed);
+
+    // on our matching json stops, sort the stops and remove duplicates
+    atcoCodesOfKnownStops = [...new Set(atcoCodesOfKnownStops.sort())];
+
+    const servicesRequiringAttention = [];
+
+    for (const key in serviceAndSortedUniqueStopsFromFeed) {
+        const serviceId = key;
+        const stops = serviceAndSortedUniqueStopsFromFeed[serviceId];
+
+        if (JSON.stringify(stops) !== JSON.stringify(atcoCodesOfKnownStops)) {
+            console.log(`The stops on ${serviceId} require attention.`);
+            servicesRequiringAttention.push(serviceId);
+        }
+    }
+
+    return servicesRequiringAttention;
+};
+
+/**
+ * For return tickets; gets a list of ids for the serices that attention
+ * because of a change on the stops relating to the service.
+ *
+ * @param {pointToPointTicket} the matching json for single ticket.
```

Comment:
Similar thing here - although return and PointToPointPeriods are the same as far as we care, we probably shouldnt just call them returns. Perhaps nonSingleTicket?